lib/plugins/aws/lib/updateStack.js

```diff
@@ -94,6 +94,24 @@ module.exports = {
       });
   },
 
+  validateStackTemplate() {
+    const compiledTemplate = _.cloneDeep(
+      this.serverless.service.provider.compiledCloudFormationTemplate);
+
+    this.serverless.cli.log('Validating Stack...');
+    const params = {
+      TemplateBody: JSON.stringify(compiledTemplate),
+    };
+
+    return this.provider.request(
+      'CloudFormation',
+      'validateTemplate',
+      params,
+      this.options.stage,
+      this.options.region
+    ).then((cfData) => this.monitorStack('validate', cfData));
```

Comment:
Yes, you basically pass the data to `monitorStack` it it will do it's thing.

`monitorStack` takes the `action` so that it can display things one the console. Furthermore it's used to do some custom stuff on a removal (since the stack might disappear after removal, but `monitoStack` still tries to query it).

The only important part to note is that the `cfData` is used to periodically query CloudFormation remotely to check the current status which is returned by AWS. Don't know if the validation is happening on your local machine or on AWS end.